### PR TITLE
Added another entry to filter field for using queryBuilder

### PR DESCRIPTION
**Pull request by symball**
For using a custom query with referenced documents and custom query

	modified:   Resources/doc/reference/filter_field_definition.rst

### CODE REVIEW

**Comment by greg0ire** [comment]
Please keep your commit subject under 50 chars (if you use vim when writing it, the color will change). If you put more information, use the § where you put the "modified" information. You can remove this information, by the way.

File: Resources/doc/reference/filter_field_definition.rst [hunk 1]
```diff
@@ -163,3 +163,34 @@ or not.
             ;
         }
     }
+
+Filter by referenced document using query
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you are attempting to filter by a referenced document, there is a chance that you will want to further configure the way the filter itself is presented.
+For example, if you have a single document controlling tags which features various classifications, you won't want all the tags to be displayed.
+Take note that in the example below, you will not be returning any kind of result from queryBuilder.
```
Comment (greg0ire) [inline Resources/doc/reference/filter_field_definition.rst]: Please wrap your lines to 80 chars (`gqq` in vim).

**Comment by symball** [comment]
Pushed new version with word wrap in place

**Comment by greg0ire** [comment]
Perfect! Now squash your commits together please.

**Comment by greg0ire** [comment]
Here is how to do that if you do not know : https://github.com/sonata-project/SonataAdminBundle/blob/master/CONTRIBUTING.md#squash-your-commits

**Comment by symball** [comment]
Thanks for the link. Rebasing something already pushed was a new one for me

**Comment by greg0ire** [comment]
Great! One last thing : use `git commit --amend` to change your commit message, then use `git push --force` again.

**Comment by greg0ire** [comment]
It did not seem to work. I would like you to keep your commit subject under 50 characters. Like this : 

```
document how to use the query builder # This is the commit bodysubject

You can add more information here if you want to.
Paragraphs are fine.  #This is the commit message body
```

**Comment by symball** [comment]
Apologies, I've just gone ahead and made the whole thing much shorter to ensure it gets through.

**Comment by greg0ire** [comment]
Great! Looks good to me!

@OskarStark @core23 can you have a final look and merge (I don't know the page bundle that well).

File: Resources/doc/reference/filter_field_definition.rst [hunk 2]
```diff
@@ -163,3 +163,38 @@ or not.
...
+
+    /* In your admin  */
+    protected function configureDatagridFilters(DatagridMapper $datagridMapper)
+    {
+        $datagridMapper
+          ->add('title')
+          ->add('author')
+          ->add('contentType', null, array(), 'document', array(
+            'class' => 'AppBundle:Tag',
+            'query_builder' => function(\AppBundle\Repository\TagRepository $dr) {
```
Comment (OskarStark) [inline Resources/doc/reference/filter_field_definition.rst]: why `$tr` ??

Comment (greg0ire) [inline Resources/doc/reference/filter_field_definition.rst]: document repository I guess

Comment (OskarStark) [inline Resources/doc/reference/filter_field_definition.rst]: should be `$tagRepository` imo

**Comment by core23** [comment]
Can't we move this to the general admin bundle?

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**